ToolBox/SamcoDataDownloader/SamcoDataDownloader.cs

```diff
@@ -0,0 +1,121 @@
+            try
+            {
+                var _samcoAPI = new SamcoBrokerageAPI();
+                _samcoAPI.Authorize(Config.Get("samco.client-id"), Config.Get("samco.client-password"), Config.Get("samco.year-of-birth"));
+
+                var castResolution = (Resolution)Enum.Parse(typeof(Resolution), resolution);
+                var castSecurityType = (SecurityType)Enum.Parse(typeof(SecurityType), securityType);
+
+                if (castSecurityType == SecurityType.Forex || castSecurityType == SecurityType.Cfd || castSecurityType == SecurityType.Crypto || castSecurityType == SecurityType.Base)
+                {
+                    throw new ArgumentException("Invalid security type: " + castSecurityType);
+                }
+
+                if (startDate >= endDate)
+                {
+                    throw new ArgumentException("Invalid date range specified");
+                }
+
+                // Load settings from config.json and create downloader
+                var dataDirectory = Globals.DataFolder;
+                var marketHoursDatabase = MarketHoursDatabase.FromDataFolder();
+                var symbolMapper = new SamcoSymbolMapper();
+
+                foreach (var pair in tickers)
+                {
+                    // Download data
+                    var pairObject = Symbol.Create(pair, castSecurityType, market);
+
+                    var exchangeTimeZone = marketHoursDatabase.GetExchangeHours(market, pairObject, castSecurityType).TimeZone;
+                    var exchange = symbolMapper.GetDefaultExchange(pairObject);
+
+                    var start = startDate.ConvertTo(DateTimeZone.Utc, exchangeTimeZone);
+                    var end = endDate.ConvertTo(DateTimeZone.Utc, exchangeTimeZone);
+
+                    // Write data
+                    var writer = new LeanDataWriter(castResolution, pairObject, dataDirectory);
+                    IList<TradeBar> fileEnum = new List<TradeBar>();
+                    var history = new List<TradeBar>();
+                    var timeSpan = new TimeSpan();
+                    switch (castResolution)
+                    {
+                        case Resolution.Tick:
+                            throw new ArgumentException("Samco Doesn't support tick resolution");
+
+                        case Resolution.Minute:
+                            history = _samcoAPI.GetIntradayCandles(pairObject.ID.Symbol, exchange, start, end).ToList();
+                            timeSpan = Time.OneMinute;
+                            break;
+
+                        case Resolution.Hour:
+                            history = _samcoAPI.GetIntradayCandles(pairObject.ID.Symbol, exchange, start, end).ToList();
+                            timeSpan = Time.OneHour;
+                            break;
+
+                        case Resolution.Daily:
+                            history = _samcoAPI.GetIntradayCandles(pairObject.ID.Symbol, exchange, start, end).ToList();
```

Comment:
`history = _samcoAPI.GetIntradayCandles(pairObject.ID.Symbol, exchange, start, end).ToList();` seems the same in all the cases of the switch? we can probably remove it from it?